exp/controllers/azuremanagedmachinepool_reconciler.go

```diff
@@ -54,23 +85,30 @@ func newAzureManagedMachinePoolReconciler(scope *scope.ManagedControlPlaneScope)
 // Reconcile reconciles all the services in pre determined order
 func (r *azureManagedMachinePoolReconciler) Reconcile(ctx context.Context, scope *scope.ManagedControlPlaneScope) error {
 	scope.Logger.Info("reconciling machine pool")
+	var normalizedVersion *string
```

Comment:
is this the same change as https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/818/files? If so let's drop it here and get it in separately with the other PR

Reply:
ok, let's pull in #818 and I'll rebase on top of it.

Reply:
rebased